backend/src/test/java/com/bakdata/conquery/integration/common/AutoConceptUtil.java

```diff
@@ -0,0 +1,106 @@
+package com.bakdata.conquery.integration.common;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.bakdata.conquery.io.cps.CPSType;
+import com.bakdata.conquery.models.datasets.Column;
+import com.bakdata.conquery.models.datasets.Table;
+import com.bakdata.conquery.models.datasets.concepts.select.Select;
+import com.bakdata.conquery.models.datasets.concepts.select.connector.DistinctSelect;
+import com.bakdata.conquery.models.datasets.concepts.select.connector.FirstValueSelect;
+import com.bakdata.conquery.models.datasets.concepts.select.connector.LastValueSelect;
+import com.bakdata.conquery.models.datasets.concepts.tree.ConceptTreeConnector;
+import com.bakdata.conquery.models.datasets.concepts.tree.TreeConcept;
+import lombok.experimental.UtilityClass;
+import org.jetbrains.annotations.NotNull;
+
+
+/**
+ * Helper class to automatically create concepts for test purposes.
+ * <p>
+ * The concept is derived from a {@link Table} and uses its name and {@link AutoConceptUtil#CONCEPT_NAME_SUFFIX}
+ * to construct an id by which it can be referenced.
+ */
+@UtilityClass
+public class AutoConceptUtil {
+
+	public static final String CONCEPT_NAME_SUFFIX = "_AUTO";
+	public static final String CONNECTOR_NAME = "connector";
+
+	/**
+	 * Generate a {@link TreeConcept} from the provided {@link Table}.
+	 * <p>
+	 * The concept will have one {@link ConceptTreeConnector} called {@link AutoConceptUtil#CONNECTOR_NAME}.
+	 * <p>
+	 * For each column in the table, a basic set of {@link Select}s is generated (compare {@link AutoConceptUtil#getAutoSelectsForColumn(Column)}).
+	 * The name of a select derives from the {@code [name]} of the column and the {@link CPSType} {@code [type]} of the {@link Select}: {@code [name]_[type]}
+	 */
+	@NotNull
+	public static TreeConcept createConcept(Table table) {
```

Comment:
Das wird ja eigentlich auf einer ReuqiredTable aufgebaut, könntest du da nicht in die RequiredTable#RequiredColumn eine Liste an Select-CPS-Type zu packen die dich interessieren? Ich finde es etwas ungeschickt den Namen einer Spalte magisch aufzuladen

Reply:
Der Name einer Resultspalte ist ja gerade z.B.: `anon_AUTO date_FIRST`. Das finde ich eigentlich Ordnung ich kann die Id vom Concept ja nicht auch `anon` nennen, weil die Tabelle schon so heißt und aus dem Selectnamen ist doch gut ersichtlich welche spalte und welcher Select gemeint ist. Für jede Spalte das extra zu konfigurieren, dann bin ich am ende doch wieder besser dranne das Concept selbst zu schreiben. Das ist dann Boilerplate.

Aber ich überlege ein Spezielles TreeConcept `AUTO_CONCEPT` zu für die Tests zu machen, dass einfach nur den Tabellennamen als parameter nimmt, und dann das selbe macht wie jetzt. Dann wäre nicht mehr das `autoConcept`-Flag da

Reply:
Ich hatte die Methode übersehen, wo du die First/Last etc händisch erstellst und dachte du machst das über eine CPS Magic. Fände es in den RequiredColumns immer noch passender weil es dann erweiterbar wäre auf alle SingleColumnSelect aber im Grunde passt das.